File: src/frontend/src/types/setupState.ts
export const SETUP_STATES = {
  READY: 'ready',
  REQUIRED: 'setup_required',
  IN_PROGRESS: 'setup_in_progress',
  FAILED: 'setup_failed',
} as const

export type SetupState = typeof SETUP_STATES[keyof typeof SETUP_STATES]

export function isSetupBlockingState(
  state: SetupState,
): state is Exclude<SetupState, typeof SETUP_STATES.READY> {
  return state !== SETUP_STATES.READY
}

File: src/informity/api/setup_state.py
"""Module for api setup state."""

from enum import StrEnum


class SetupState(StrEnum):
    """Class docstring."""
    READY = "ready"
    REQUIRED = "setup_required"
    IN_PROGRESS = "setup_in_progress"
    FAILED = "setup_failed"


__all__ = ["SetupState"]
